osf/models/user.py

```diff
@@ -695,6 +696,8 @@ def merge_user(self, user):
                 node.contributor_set.filter(user=user).delete()
             else:
                 node.contributor_set.filter(user=user).update(user=self)
+                node.add_permission(self, user_perms)
+                node.remove_permission(user, user_perms)
```

Comment:
Changes for merging users, need to point the contributor object to the new contributor and remove the old permissions from the user.